pkg/util/utils.go

```diff
@@ -14,18 +14,27 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package validation
+package utils
 
 import (
 	gatewayv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
 )
 
-func pathMatchTypePtr(s string) *gatewayv1a2.PathMatchType {
+// PathMatchTypePtr translates a string to *PathMatchType
+func PathMatchTypePtr(s string) *gatewayv1a2.PathMatchType {
+	if s != string(gatewayv1a2.PathMatchExact) && s != string(gatewayv1a2.PathMatchPrefix) && s != string(gatewayv1a2.PathMatchRegularExpression) &&
+		s != string(gatewayv1a2.PathMatchImplementationSpecific) {
+		return nil
+	}
```

Comment:
I'm not sure we want to be doing validation here. It's possible that we may want to use this func to supply an invalid value for a test case.

Reply:
b1e87b89841a036ad7108d6cb0c50fa65e4c3990